ext/fg/js/frontend.js

```diff
@@ -41,15 +35,23 @@ class Frontend extends TextScanner {
 
         this._disabledOverride = false;
 
-        this.options = null;
+        this._options = null;
 
         this._pageZoomFactor = 1.0;
         this._contentScale = 1.0;
         this._orphaned = false;
         this._lastShowPromise = Promise.resolve();
 
+        this._enabledEventListeners = new EventListenerCollection();
+        this._textScanner = new TextScanner(
+            window,
+            () => this.popup.isProxy() ? [] : [this.popup.getContainer()],
+            [(x, y) => this.popup.containsPoint(x, y)]
+        );
+        this._textScanner.onSearchSource = this.onSearchSource.bind(this);
```

Comment:
It looks a bit inconsistent that the synchronous listener(s) use `this._textScanner.on(...)` but this asynchronous one is overridden. This is probably the simplest way to do it, but just mentioning

Reply:
I agree; I was trying to make the switch with minimal changes, and I think modifying `onSearchSource` would have been a bit more involved. Mainly because it affects `QueryParser` too. Eventually that class can be updated as well.